text/0039-reduce-module-size.md

```diff
@@ -0,0 +1,533 @@
+# Reduce aws-cdk-lib package size
+
+* **Original Author(s):**: @madeline-k
+* **Tracking Issue**: #39
+* **API Bar Raiser**: tbd
+
+The AWS CDK v2 framework library, `aws-cdk-lib` contains all packages from AWS
+CDK v1 in one monolithic package. While this is a great value for customers in
+making dependency management much easier, it has also resulted in increasing the
+unpacked npm package size to 234 MB, at the time of writing. This feature
+reduces the size of the package to under 100 MB.
+
+## Working Backwards
+
+* **CHANGELOG**:
+
+feat(aws-cdk-lib): Reduce aws-cdk-lib package size.
+
+‼️ If you use any of the following Constructs in a network-restricted
+environment, you might encounter problems upgrading to this version. ‼️
+
+* lambda_layer_kubectl.KubectlLayer
+* aws_eks.KubectlProvider
+* aws_eks.HelmChart
+* aws_eks.KubernetesManifest
+* aws_eks.KubernetesObjectValue
+* aws_eks.KubernetesPatch
+* lambda_layer_awscli.AwsCliLayer
+* aws_s3_deployment.BucketDeploymentß
+* aws_stepfunctions_tasks.EmrContainersStartJobRun
+* lambda_layer_node_proxy_agent.NodeProxyAgentLayer
+* aws_eks.FargateProfile
+* aws_eks.Cluster
+
+Please see [#1234](https://github.com/aws/aws-cdk/issues/1234) for more details,
+and comment on the issue if you run into problems.
+
+* **CLI Notices**:
+
+We can deliver a notice to customers based on what specific constructs they are
+using. We will define separate notices for each different scenario.
+
+Title: (aws-cdk-lib): upcoming change in packaging structure might require
+action to upgrade
+
+Body: We’ve identified that you are using at least one of the [List of
+Constructs] Constructs. If you are running your aws-cdk commands in an
+environment that does not have access to npm, then you will need to make a
+change to make version 2.x of [npm packages] available in your environment. See
```

Comment:
What change is this? I feel this summary could already give some hint on what the user would need to do. Otherwise this sounds super scary before you actually read the issue.